full-stack-intro/client/src/components/Todos.tsx

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+/* eslint-disable @typescript-eslint/no-empty-function -- Remove me */
+import { useEffect, useState } from 'react';
+import { PageTitle } from './PageTitle';
+import { TodoList } from './TodoList';
+import { TodoForm } from './TodoForm';
+
+export type UnsavedTodo = {
+  task: string;
+  isCompleted: boolean;
+};
+export type Todo = UnsavedTodo & {
+  todoId: number;
+};
+
+export function Todos() {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+
+  /* Implement useEffect to fetch all todos. Hints are at the bottom of the file. */
+  useEffect(() => {
+    async function work() {
+      try {
+        const response = await fetch('/api/todos');
+        if (!response.ok) throw new Error('Response get method not ok.');
+        const data = await response.json();
+        setTodos(data);
+      } catch (error) {
+        setError(error);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    work();
+  }, []);
+
+  /* Implement addTodo to add a new todo. Hints are at the bottom of the file. */
+  async function addTodo(newTodo: UnsavedTodo) {
+    async function postData() {
```

Comment:
The nested async function is only needed in useEffect. It is not needed in event handler functions.